src/DependencyInjection/Compiler/ReorderTranslationResourceFilesPass.php

```diff
@@ -0,0 +1,38 @@
+<?php
+/**
+ * @copyright Zicht Online <https://zicht.nl>
+ */
+
+namespace Zicht\Bundle\MessagesBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class ReorderTranslationResourceFilesPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        /* This Compiler Pass is searching for .db translation files to place them at the end
+           This will cause the Message Catalogue to load these resources as last and thus
+           the effect of overriding messages is maintained
+         */
+        $translator = $container->findDefinition('translator.default');
+        $options = $translator->getArgument(4);
+        $resourceFilesByLocale =& $options['resource_files'];
+
+        foreach ($resourceFilesByLocale as $locale => $files) {
+            $dbFiles = [];
+            foreach ($files as $key => $file) {
+                if (substr($file, -3) === '.db') {
+                    $dbFiles[$key] = $file;
+                    unset($resourceFilesByLocale[$locale][$key]);
+                }
+            }
+            if (count($dbFiles) > 0) {
+                $resourceFilesByLocale[$locale] = array_merge($resourceFilesByLocale[$locale], $dbFiles);
+            }
+        }
```

Comment:
Zou dit niet makkelijker kunnen met een `usort` oid?  De sorting function geeft dan 1 terug voor files met `.db` en 0 voor alle andere files... leest meer als 'sorting' dan een array unset en daarna array merge, en is waarschijnlijk ook korter